dvc/config.py

```diff
@@ -468,6 +468,19 @@ def validate(data):
             raise ConfigError(str(exc)) from None
 
 
+def _format_config(config, parent=None):
+    for section, options in config.items():
+        for option, value in options.items():
+            key = f"{section}.{option}"
+            if parent is not None:
+                key = f"{parent}.{key}"
+
+            if isinstance(value, dict):
+                yield from _format_config({option: value}, parent=section)
+            else:
+                yield f"{key}={value!r}"
```

Comment:
Do you remember which objects in config required repr instead of str?

Reply:
It was just for visual convenience (distinguishing strings and other values). FWIW `git config --list` uses raw values,
```
user.name=Batuhan Taskaya
user.email=batuhanosmantaskaya@gmail.com
user.signingkey=C3ABD12D
core.excludesfile=/home/isidentical/.gitignore
core.repositoryformatversion=0
core.filemode=true
```

I also noticed that this is a bit inconsistent with the behavior of `dvc config <name>`, since it doesn't use repr. So, I assume if the `repr()` is fine, it would be nice to change the output of `dvc config <name>` too! (or the other way around.)

Reply:
@isidentical Ah, makes sense. Let's use str here then, since `dvc config section.option` uses `str` and `git config --local` also represents the strings the same way. Just for consistency.